source/retryable-writes/retryable-writes.rst

```diff
@@ -511,6 +513,15 @@ experience across all drivers.
 Why expose users to server-side errors for unsupported operations?
 ------------------------------------------------------------------
 
+.. todo:: Determine how this should change. Do we end up using whitelists for
+   supported operations (based on wire protocol verison) and avoid injecting
+   transaction IDs on all write ops, or rely on the server to raise an error if
+   unsupported ops would be retried? In both cases, it's possible for users to
+   be unaware that some operations are not supported until some error in prod.
+   Given the MongoClient-wide scope of retryWrites, we likely don't want to
+   prohibit unsupported operations outright (either in the client API or by
+   having the server raise an exception on the first attempt).
+
```

Comment:
Apparently it is *not* a goal now to help users realized that some operations cannot be retried. We've abandoned that in favor of this goal: make it possible to turn on retryability for a subset of operations, with minimum code changes. Therefore, we must allow unacknowledged writes even when retryWrites is true.

For acknowledged writes that aren't retryable on the server, there's a problem. The server can't *know* whether it's already executed any particular operation or not. E.g., I send an updateMany with a txnNum, and I can't read the server response. Two options:

1. I send it again. The server does not store a transaction record for a non-retryable operation, so it doesn't know whether it's executed it before. Therefore the server may execute the updateMany twice.
2. The driver maintains a whitelist (sorry) of retryable operations and only re-sends operations that are in the whitelist.